src/android/CustomFCMReceiverPlugin.java

```diff
@@ -0,0 +1,155 @@
+import org.apache.cordova.CordovaWebView;
+import org.apache.cordova.firebase.FirebasePluginMessageReceiver;
+import com.dmarc.cordovacall.MyConnectionService;
+import com.google.firebase.messaging.RemoteMessage;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+import java.lang.reflect.Method;
+
+public class CustomFCMReceiverPlugin {
+    static final String TAG = "CustomFCMReceiverPlugin";
+    private CustomFCMReceiver customFCMReceiver;
+    private TelecomManager tm;
+    private PhoneAccount phoneAccount;
+    private PhoneAccountHandle handle;
+    private String appName;
+    private String from;
+    private Context applicationContext;
+
+    public void initialize(Context initialApplicationContext) {
+        Log.d(TAG, "initialize");
+        try {
+            Log.d(TAG, "initialApplicationContext: " + initialApplicationContext.toString());
+            applicationContext = initialApplicationContext;
+            customFCMReceiver = new CustomFCMReceiver();
+            appName = getApplicationName();
+            tm = (TelecomManager) applicationContext.getSystemService(Context.TELECOM_SERVICE);
+            handle = getExistingPhoneAccountHandle();
+        } catch (Exception e) {
+            handleException("Initializing plugin", e);
+        }
+    }
+
+    private String getApplicationName() {
+        ApplicationInfo applicationInfo = applicationContext.getApplicationInfo();
+        int stringId = applicationInfo.labelRes;
+        return stringId == 0 ? applicationInfo.nonLocalizedLabel.toString() : applicationContext.getString(stringId);
+    }
+
+    private PhoneAccountHandle getExistingPhoneAccountHandle() {
+        for (PhoneAccountHandle accountHandle : tm.getCallCapablePhoneAccounts()) {
+            PhoneAccount phoneAccount = tm.getPhoneAccount(accountHandle);
+            if (phoneAccount != null && phoneAccount.getLabel().toString().equals(appName)) {
+                return accountHandle;
+            }
+        }
+        return null;
+    }
+
+    protected static void handleError(String errorMsg) {
+        Log.e(TAG, errorMsg);
+    }
+
+    protected static void handleException(String description, Exception exception) {
+        handleError(description + ": " + exception.toString());
+    }
+
+    private Map<String, String> bundleToMap(Bundle extras) {
+        Map<String, String> map = new HashMap<String, String>();
+
+        Set<String> ks = extras.keySet();
+        Iterator<String> iterator = ks.iterator();
+        while (iterator.hasNext()) {
+            String key = iterator.next();
+            map.put(key, extras.getString(key));
+        }
+        return map;
+    }
+
+    private boolean inspectAndHandleMessageData(Map<String, String> data) {
+        boolean isHandled = false;
+        Log.d(TAG, "Inspecting message.");
+        Log.d(TAG, data.toString());
+
+        if (data.containsKey("callType")) {
```

Comment:
This would be wrong or outdated, based on the facetalk job: SoftphoneFcmInviteMessageJob the format of the data would be:

<img width="975" height="52" alt="Screenshot from 2025-09-26 12-26-23" src="https://github.com/user-attachments/assets/e20add11-29db-48b6-b5bb-858660102c40" />

Don't worry about changing this branch / PR i'm going to create my own